api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -337,6 +337,22 @@ public void getWorkspaces_skipDeleted() {
     assertThat(workspaceService.getWorkspaces().size()).isEqualTo(currentWorkspacesSize);
   }
 
+  @Test
+  public void getWorkspaces_skipPublished() {
+    int currentWorkspacesSize = workspaceService.getWorkspaces().size();
+    workbenchConfig.featureFlags.enablePublishedWorkspacesViaDb = true;
+    DbWorkspace mockWorkspace =
+        buildDbWorkspace(
+            workspaceIdIncrementer.getAndIncrement(),
+            "published",
+            DEFAULT_WORKSPACE_NAMESPACE,
+            WorkspaceActiveStatus.ACTIVE);
+    mockWorkspace.setPublished(false);
+    mockWorkspace.setFeaturedCategory(DbFeaturedCategory.TUTORIAL_WORKSPACES);
+    addMockedWorkspace(mockWorkspace);
+    assertThat(workspaceService.getWorkspaces().size()).isEqualTo(currentWorkspacesSize);
```

Comment:
I am not clear why the workspace created in this test is skipped.

In [addMockedWorkspace](https://github.com/all-of-us/workbench/blob/main/api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java#L295), it seems like the workspace is saying that the user is the owner. My understanding is that even if a workspace is featured/published, it is still shown if the user is an owner or writer.

Would it be helpful to have different tests for owner/writer/viewer?

Reply:
The workspaceService test were not working as expected as the firecloudMapper was Mocked therefore it always used to return null for 
fcResponse.accessLevel in workspaceService. This has been fixed and added a new test to ensure publishedWorkspace with OWNER is not skipped